x-pack/plugin/ingest/src/main/java/org/elasticsearch/xpack/ingest/UrlPartsProcessor.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ingest;
+
+import org.elasticsearch.ingest.AbstractProcessor;
+import org.elasticsearch.ingest.ConfigurationUtils;
+import org.elasticsearch.ingest.IngestDocument;
+import org.elasticsearch.ingest.Processor;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UrlPartsProcessor extends AbstractProcessor {
+
+    public static final String TYPE = "url_parts";
+
+    private final String field;
+    private final String targetField;
+    private final boolean removeIfSuccessful;
+    private final boolean keepOriginal;
+
+    UrlPartsProcessor(String tag, String description, String field, String targetField, boolean removeIfSuccessful, boolean keepOriginal) {
+        super(tag, description);
+        this.field = field;
+        this.targetField = targetField;
+        this.removeIfSuccessful = removeIfSuccessful;
+        this.keepOriginal = keepOriginal;
+    }
+
+    public String getField() {
+        return field;
+    }
+
+    public String getTargetField() {
+        return targetField;
+    }
+
+    public boolean getRemoveIfSuccessful() {
+        return removeIfSuccessful;
+    }
+
+    public boolean getKeepOriginal() {
+        return keepOriginal;
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
+        String value = ingestDocument.getFieldValue(field, String.class);
+
+        URL url;
+        try {
+            url = new URL(value);
```

Comment:
Are there limitations on which schemes that this can parse? Would these parse?

- ftp://ftp.is.co.za/rfc/rfc1808.txt
- ldap://[2001:db8::7]/c=GB?objectClass?one
- telnet://192.0.2.16:80/

Perhaps using java.net.URI would be more forgiving and not require a URLStreamHandler to be loaded.

Reply:
Similarly parts of the URL are required to for parsing to work?

Reply:
@andrewkroh, thanks for looking it over and commenting. I switched over to `java.net.URI` which does support more schemes including all three of the examples you list above.

Right now, no parts of a URI are required beyond what `java.net.URI` needs to construct an instance. Is that what you would prefer?